extras/ioc_hash_scan.sh

```diff
@@ -0,0 +1,189 @@
+# export AWS_REGION="us-east-1" >> "$HOME/.bashrc"
+# Then reload your shell, like
+# source ~/.bashrc
+#
+# This script assumes that it exists in the ioc-scanner/extras/ directory.
+# If this is different, please edit the variable $pydir to point to
+# the directory containing ioc_scanner.py
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Directory path of the ioc_scanner.py
+pydir="../src/ioc_scan/"
+
+if [ $# -eq 0 ] || [ "$1" = "-h" ] || [ "$1" = "--help" ] || [ $# -lt 2 ]; then
+  echo Usage: "$0" instance-list-file \<AWS_PROFILE\>
+  exit 1
+fi
+
+# Check if instance list file exists.
+if [ ! -f "$1" ]; then
+  echo Instance List file "$1" does not exist - exiting.
+  exit 1
+fi
+
+# Read instance id strings from file.  [[ -n "$line" ]] handles the case where
+# the last line doesn't end with a newline.
+serverList=()
+while IFS= read -r line || [[ -n "$line" ]]; do
+  serverList+=("$line")
+done < "$1"
+
+if [ ${#serverList[@]} -eq 0 ]; then
+  echo No Instances found in "$1" - exiting.
+  exit 1
+fi
+
+# Path of aws credentials file
+AWSPROF="$2"
+
+today=$(date +%Y-%m-%d)
+logfile="$HOME/$today-ioc-scanner-hashscan.log"
+
+# Suppress some verbose stdout.
+# Suppress stnderr of pkill command
+exec > >(grep --invert-match 'Starting\|Exiting')
+exec 2> >(grep --invert-match 'SIGTERM')
+
+## FUNCTIONS
+function getOSType() {
+  OSNAME=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=cat /etc/os-release | grep --extended-regexp ^NAME=" \
+    --output text)
+  echo "$OSNAME" | grep NAME | cut -d'"' -f2 | cut -d' ' -f1
+}
+
+function getHost() {
+  HOST=$(AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartInteractiveCommand \
+    --parameters="command=hostname" \
+    --output text)
+  echo "$HOST" | grep --invert-match "session" | sed '/^$/d'
+}
+
+function portForward() {
+  AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+    aws --profile="$AWSPROF" --region="$AWS_REGION" \
+    ssm start-session --target="$i" \
+    --document=AWS-StartPortForwardingSession --parameters="localPortNumber=5555,portNumber=6666"
+}
+
+function installNC() {
+  if [[ "$OS" == "Debian" ]]; then
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo apt-get install netcat -y'"
+  else
+    AWS_SHARED_CREDENTIALS_FILE="$AWS_CREDENTIALS_FILE" \
+      aws --profile="$AWSPROF" --region="$AWS_REGION" \
+      ssm start-session --target="$i" \
+      --document=AWS-StartInteractiveCommand \
+      --parameters="command='sudo yum install netcat -y'"
```

Comment:
Modern RH-based systems use `dnf`, not `yum`.
```suggestion
      --parameters="command='sudo dnf --assumeyes install netcat'"
```